docs/src/main/asciidoc/security-keycloak-authorization.adoc

```diff
@@ -417,6 +387,97 @@ quarkus.keycloak.policy-enforcer.paths.1.enforcement-mode=ENFORCING
+quarkus.keycloak.policy-enforcer.paths.1.name=Scope Permission Resource
+quarkus.keycloak.policy-enforcer.paths.1.path=/api/protected/standard-way
+quarkus.keycloak.policy-enforcer.paths.1.methods.get.method=GET
+quarkus.keycloak.policy-enforcer.paths.1.methods.get.scopes=read <1>
+
+# path policies without scope
+quarkus.keycloak.policy-enforcer.paths.2.name=Scope Permission Resource
+quarkus.keycloak.policy-enforcer.paths.2.path=/api/protected/programmatic-way
+quarkus.keycloak.policy-enforcer.paths.3.name=Scope Permission Resource
+quarkus.keycloak.policy-enforcer.paths.3.path=/api/protected/annotation-way
+----
+<1> User must have resource permission 'Scope Permission Resource' and scope 'read'
+
+Request path `/api/protected/standard-way` is now secured by the Keycloak Policy Enforcer and does not require
+any additions (such as `@RolesAllowed` annotation). In some cases, you may want to perform the same check programmatically.
+You are allowed to do that by injecting a `SecurityIdentity` instance in your beans, as demonstrated in the example below.
+Alternatively, if you annotate resource method with the `@PermissionsAllowed` annotation, you can achieve the same effect.
+The following example shows three resource method that all requires same 'read' scope:
+
+[source,java]
+----
+import static io.quarkus.security.PermissionsAllowed.PERMISSION_TO_ACTION_SEPARATOR;
+
+import java.security.BasicPermission;
+import java.util.List;
+
+import jakarta.inject.Inject;
+import jakarta.ws.rs.ForbiddenException;
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.Path;
+
+import org.keycloak.representations.idm.authorization.Permission;
+
+import io.quarkus.security.PermissionsAllowed;
+import io.quarkus.security.identity.SecurityIdentity;
+import io.smallrye.mutiny.Uni;
+
+@Path("/api/protected")
+public class ProtectedResource {
+
+    public static final String REQUIRED_PERMISSION = "Scope Permission Resource";
+    public static final String REQUIRED_ACTION = "read";
+
+    @Inject
+    SecurityIdentity identity;
+
+    @GET
+    @Path("/standard-way")
+    public Uni<List<Permission>> standardWay() { <1>
+        return Uni.createFrom().item(identity.<List<Permission>> getAttribute("permissions"));
+    }
+
+    @GET
+    @Path("/programmatic-way")
+    public Uni<List<Permission>> programmaticWay() {
+        var requiredPermission = new BasicPermission(REQUIRED_PERMISSION) {
+            @Override
+            public String getActions() {
+                return REQUIRED_ACTION;
+            }
+        };
+        return identity.checkPermission(requiredPermission).onItem() <2>
+                .transform(granted -> {
+                    if (granted) {
+                        return identity.getAttribute("permissions");
+                    }
+                    throw new ForbiddenException();
+                });
+    }
+
+    @PermissionsAllowed(REQUIRED_PERMISSION + PERMISSION_TO_ACTION_SEPARATOR + REQUIRED_ACTION) <3>
```

Comment:
Can you please use the actual values here as it is a bit difficult to follow what the permission value is

Reply:
Sure, I though this way, it will be stressed that constants are same for programmatic approach and annotation one. Will do.